src/routers/event.router.ts

```diff
@@ -1,3 +1,91 @@
-import { Router } from "express";
-
+import { Router, type Request, type Response } from "express";
+import {
+  createEvent,
+  getAllEvents,
+  getEventByCode,
+  addStaffToEvent,
+  addYouthToEvent,
+} from "../controllers/event.controller";
 export const eventRouter = Router();
+
+//review: is it good to return raw errors to the frontend?
+//review: error codes: how to differentiate between 500 and 404 if both will cause mongoose to throw error?
+
+//gets list of all events
+eventRouter.get("/", async (_req: Request, res: Response) => {
+  try {
+    const allEvents = await getAllEvents();
+    res.status(200).json(allEvents);
+  } catch (err) {
+    //typescript makes us do this
+    if (err instanceof Error) {
+      res.status(404).json({ error: err.message });
+    } else {
+      res.status(404).json({ error: "An unknown error occurred" });
+    }
+  }
+});
+
+//gets event with a provided code
+eventRouter.get("/:eventCode", async (req, res) => {
+  try {
+    //technically an array of events could be returned
+    const event = await getEventByCode(req.params.eventCode);
+    res.status(200).json(event);
+  } catch (err) {
+    //typescript makes us do this
+    if (err instanceof Error) {
+      res.status(404).json({ error: err.message });
+    } else {
+      res.status(404).json({ error: "An unknown error occurred" });
+    }
+  }
+});
+
+//adds event
+eventRouter.post("/", async (req: Request, res: Response) => {
+  try {
+    let event = await createEvent(req.body);
+    res.status(200).json(event);
+  } catch (err) {
+    //typescript makes us do this
+    if (err instanceof Error) {
+      res.status(500).json({ error: err.message });
+    } else {
+      res.status(500).json({ error: "An unknown error occurred" });
+    }
+  }
+});
+
+//adds a staff member to an event
+eventRouter.put("/addStaff/:eventCode", async (req, res) => {
+  try {
+    let event = await addStaffToEvent(
+      req.params.eventCode,
+      req.body.firebaseUID,
+    );
+    res.status(200).json(event);
+  } catch (err) {
+    //typescript makes us do this
+    if (err instanceof Error) {
+      res.status(500).json({ error: err.message });
+    } else {
+      res.status(500).json({ error: "An unknown error occurred" });
+    }
+  }
+});
+
+//adds a youth to an event
+eventRouter.put("/attend/:eventCode", async (req, res) => {
+  try {
+    await addYouthToEvent(req.params.eventCode, req.body.firebaseUID);
+    res.status(200).send("Youth marked as present"); //review: would it be more useful to just return the updated document
```

Comment:
I think this should be fine since we don't need to return the entire event object when a youth is marked as attended.